cli_tools/gce_ovf_import/ovf_importer/ovf_param_validator.go

```diff
@@ -0,0 +1,255 @@
+	}
+
+	if params.InstanceNames != "" && params.MachineImageName != "" {
+		return daisy.Errf("-%v and -%v can't be provided at the same time", InstanceNameFlagKey, MachineImageNameFlagKey)
+	}
+
+	if params.IsInstanceImport() {
+		// instance import specific validation
+		instanceNameSplits := strings.Split(params.InstanceNames, ",")
+		if len(instanceNameSplits) > 1 {
+			return daisy.Errf("OVF import doesn't support multi instance import at this time")
+		}
+
+		if params.MachineImageStorageLocation != "" {
+			return daisy.Errf("-%v can't be provided when importing an instance", MachineImageStorageLocationFlagKey)
+		}
+	}
+
+	if err := validation.ValidateStringFlagNotEmpty(params.OvfOvaGcsPath, OvfGcsPathFlagKey); err != nil {
+		return err
+	}
+
+	if err := validation.ValidateStringFlagNotEmpty(params.ClientID, ClientIDFlagKey); err != nil {
+		return err
+	}
+
+	if _, err := storageutils.GetBucketNameFromGCSPath(params.OvfOvaGcsPath); err != nil {
+		return daisy.Errf("%v should be a path to OVF or OVA package in Cloud Storage", OvfGcsPathFlagKey)
+	}
+
+	if params.Labels != "" {
+		var err error
+		params.UserLabels, err = param.ParseKeyValues(params.Labels)
+		if err != nil {
+			return err
+		}
+	}
+
+	if params.NodeAffinityLabelsFlag != nil {
+		var err error
+		params.NodeAffinities, params.NodeAffinitiesBeta, err = compute.ParseNodeAffinityLabels(params.NodeAffinityLabelsFlag)
+		if err != nil {
+			return err
+		}
+	}
+
+	if params.Hostname != "" {
+		err := validation.ValidateFqdn(params.Hostname, HostnameFlagKey)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *ParamValidatorAndPopulator) determineProject(originalProject string) (*string, error) {
+	project, err := param.GetProjectID(p.metadataClient, strings.TrimSpace(originalProject))
+	return &project, err
+}
+
+func (p *ParamValidatorAndPopulator) determineRegion(zone string) (string, error) {
+	zoneSplits := strings.Split(zone, "-")
+	if len(zoneSplits) < 2 {
+		return "", daisy.Errf("%v is not a valid zone", zone)
+	}
+	return strings.Join(zoneSplits[:len(zoneSplits)-1], "-"), nil
+}
+
+func (p *ParamValidatorAndPopulator) determineZone(project, originalZone string) (zone string, err error) {
+	zone = strings.TrimSpace(originalZone)
+	if zone != "" {
+		if err := p.zoneValidator.ZoneValid(project, zone); err != nil {
+			return "", err
+		}
+		return zone, nil
+	}
+	if !p.metadataClient.OnGCE() {
+		return "", fmt.Errorf("zone cannot be determined because build is not running on GCE")
+	}
+	// determine zone based on the zone Cloud Build is running in
+	zone, err = p.metadataClient.Zone()
+	if err != nil || zone == "" {
+		return "", fmt.Errorf("can't infer zone: %v", err)
+	}
+	return zone, nil
+}
+
+func (p *ParamValidatorAndPopulator) determineScratchBucket(originalBucket, project, region, buildID string) (scratchBucket string, err error) {
```

Comment:
This is doing more than determining the path to a scratch bucket, as it's also creating it. The function name should reflect that.

Reply:
Renamed to `createScratchBucketIfMissing`